lib/extensions/ar_set_application_name.rb

```diff
@@ -0,0 +1,14 @@
+module ArSetApplicationName
+  # Sets the current and future db connections' application
+  # name if the name is not known until after some startup
+  # and querying some tables.
+  #
+  # If we can determine the name before establish_connection
+  # time, it's probably better to module prepend that method
+  # and run "SET application_name TO 'my_app'"
+  def self.name=(name)
+    # TODO: this is postgresql specific
+    ENV['PGAPPNAME'] = name
+    ActiveRecord::Base.connection_pool.disconnect!
```

Comment:
We set this ENV variable and disconnect because:
* we probably have already established at least one connection
* any new thread/connection needs to use this value

Unfortunately, we need to query the db to determine the name so we can't module prepend establish_connection (and call super).